### PR TITLE
fix(knowbase): search article with multi categories

### PR DESCRIPTION
**Pull request by Rom1-B**
When an article belongs to several categories and you filter on the category in the search, in the tree you have the counter on the categories excluded by the filter, but no result when you open the category.

Before:
image

After:
image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: src/Features/TreeBrowse.php [hunk 1]
```diff
@@ -203,13 +203,25 @@ public static function getTreeCategoryList(string $itemtype, array $params): arr
         $cat_fk    = $cat_itemtype::getForeignKeyField();
         $cat_join = $itemtype . '_' . $cat_itemtype;
 
+        // This query is used to get the IDs of all results matching the search criteria
+        $sql = $data['sql']['search'];
+        // We can remove all the SELECT fields and replace it with just the ID field
+        $raw_select = $data['sql']['raw']['SELECT'];
```
Comment (trasher) [inline src/Features/TreeBrowse.php:209]:
```suggestion
```
Since this is already done just before

**Review by trasher — APPROVED** [review]
I'm not at all fan of replacement solution, but since this is based on Search; I think we currently have no other solution. ALso, this is already done just like this just before; so I guess this is OK.

**Review by trasher — APPROVED** [review]

**Comment by stonebuzz** [comment]
When a search does not contains criteria about "category" an error SQL occurs

image

**Comment by AdrienClairembault** [comment]
FYI, this break searches on documents:

image

**Review by cedric-anne — APPROVED** [review]
Seems OK, but it is a real pain to read and would probably a pain to maintain.

**Review by trasher — APPROVED** [review]
LGTM, but I'm very unsure :D

**Review by AdrienClairembault — APPROVED** [review]

**Pull request merged**